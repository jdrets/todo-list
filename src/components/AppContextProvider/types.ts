import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
