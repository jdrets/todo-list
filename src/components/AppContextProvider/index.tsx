import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
