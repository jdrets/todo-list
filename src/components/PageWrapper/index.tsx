import React from 'react'

import {
  Wrapper
} from './styles'

const PageWrapper = ({ children }) => (
  <Wrapper>{children}</Wrapper>
)

export default PageWrapper
