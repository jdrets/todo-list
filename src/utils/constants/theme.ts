const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
