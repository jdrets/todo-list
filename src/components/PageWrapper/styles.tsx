import styled from 'styled-components'

import theme from '../../utils/constants/theme'

export const Wrapper = styled.section`
  background-color: white;
  border-radius: 60px;
  min-height: calc(100vh - 210px);
  margin-top: -60px;
  padding: 24px;

  @media (min-width: ${theme.layout.breakpoints.medium}) {
    border-radius: 0;
    height: calc(100vh - 150px);
    margin: 0 auto;
    width: 600px;
  }
`

export default Wrapper
