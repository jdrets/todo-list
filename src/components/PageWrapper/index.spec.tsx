import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'

import theme from '../../utils/constants/theme'
import AppContextProvider from '../AppContextProvider'

import PageWrapper from '.'

const setup = () => (
  render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <PageWrapper>Main wrapper</PageWrapper>
      </AppContextProvider>
    </ThemeProvider>
  )
)

describe('<PageWrapper />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
