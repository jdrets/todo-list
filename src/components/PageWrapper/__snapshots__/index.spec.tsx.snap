// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapper /> should match with snapshot 1`] = `
<div>
  <section
    class="sc-bdnylx gHaPUZ"
  >
    Main wrapper
  </section>
</div>
`;
